### PR TITLE
refactor: add integration config for tornado, refactor distributed tracing helper usage

### PR DESCRIPTION
**Pull request by Yun-Kim**
## Description
From #2223, it was found that the tornado integration did not declare its own integration config and instead relied on storing distributed_tracing in the app settings object. To unify integrations to follow the same pattern, an integration config was added to tornado with the distributed_tracing setting set to True by default. Additionally, the `activate_distributed_headers` helper from `trace_utils` was used to refactor the distributed tracing in tornado handlers.

### CODE REVIEW

**Comment by Yun-Kim** [comment]
To do: consider if settings dictionary in `tornado/application.py` can be refactored to use the integration config object

File: releasenotes/notes/tornado-int-config-a8900a60201d4d61.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+other:
+  - |
+    Added distributed tracing setting to integration config for tornado.
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/tornado-int-config-a8900a60201d4d61.yaml:4]: Need to be a bit more specific again

File: ddtrace/contrib/tornado/patch.py [hunk 2]
```diff
@@ -9,9 +10,19 @@
...
 from . import template
+from ...utils.formats import asbool
+from ...utils.formats import get_env
 from ...utils.wrappers import unwrap as _u
 
 
+config._add(
+    "tornado",
+    dict(
+        distributed_tracing=asbool(get_env("tornado", "distributed_tracing", default=True)),
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/tornado/patch.py:21]: Need tests for this

**Review by Kyle-Verhoog — CHANGES_REQUESTED** [review]

**Comment by Yun-Kim** [comment]
> For this reason, I don't want to add the env variable unless we're actively going to remove the `settings.distributed_tracing` option.

My solution for this was to make `settings.distributed_tracing`  default to `None`, and use the environment variable `DD_TORNADO_DISTRIBUTED_TRACING` default to `True` and referenced by `config.tornado.distributed_tracing`. This way, nothing functionally changes as it will by default propagate distributed traces, but now will rely on the int_config by default. If any user makes any changes to their `settings.distributed_tracing`, this will override the int_config and env var.

File: ddtrace/contrib/tornado/__init__.py [hunk 3]
```diff
@@ -96,8 +96,9 @@ def log_exception(self, typ, value, tb):
 * ``tags`` (default: `{}`): set global tags that should be applied to all spans.
 * ``enabled`` (default: `True`): define if the tracer is enabled or not. If set to `false`, the
   code is still instrumented but no spans are sent to the APM agent.
-* ``distributed_tracing`` (default: `True`): enable distributed tracing if this is called
-  remotely from an instrumented application.
+* ``distributed_tracing`` (default: `None`): enable distributed tracing if this is called
+  remotely from an instrumented application. Overrides the integration config which is configured via the
+  ``DD_TORNADO_DISTRIBUTED_TRACING` environment variable.
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/tornado/__init__.py:101]: :+1:

**Review by Kyle-Verhoog — APPROVED** [review]
nice job! lgtm

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**